perma_web/perma/celery_tasks.py

```diff
@@ -1257,12 +1257,19 @@ def convert_warc_to_wacz(input_guid, benchmark_log):
     warc_path = f"{link.guid}.warc.gz"
     wacz_path = f"{link.guid}.wacz"
     pages_path = "pages.jsonl"
+    warc_not_found_error = False
 
     # save a local copy of the warc file
-    with open(warc_path, "wb") as file:
-        copy_file_data(default_storage.open(link.warc_storage_file()), file)
+    warc_save_start_time = time.time()
+    try:
+        with open(warc_path, "wb") as file:
+            copy_file_data(default_storage.open(link.warc_storage_file()), file)
+    except FileNotFoundError:
```

Comment:
I think this situation should never come up: we should never run this task on Perma Links that failed and therefore lack WARCs. We have a data problem on stage, where the field that should indicate whether a Perma Link can be played back or not is sometimes wrong, but we confirmed that that is _not_ a problem on prod (and shouldn't be a problem in anybody's dev database). So I think we probably do not need to account for FileNotFound errors.

Are you thinking it is just good "defensive coding"? If yes, how about, instead of failing here with FileNotFoundError, instead doing some validation of the Link at the beginning, for instance, making sure that `cached_can_play_back` is True?

Reply:
Yep, it was mostly defensive, but I will just remove it since it sounds like we won't run into this in prod.

Reply:
Cool beans.